### DISCUSSION TITLE
Is there a way to search nested data?

### DISCUSSION THREAD

**Discussion by piersoh** [saulpw/visidata#1049] [Q&A]
I've found visidata gerat for looking at data files, but I've got some json files with nested data - I wondered if there was simple way to search all the data even if it's nested? My current alternative is just open the file in vim/less and search it that way.

**Comment by geekscrapy** [comment]
No way to really search embedded structures per-say, however:

If the json is contained within a single cell and not large you could make that column a string type and then search from there? So load the file, select the column you want then `type-string`. The. Use the search type you want, likely `select-col-regex`

Is it json or jsonl you have?

**Reply by piersoh** [reply]
Thanks for getting back to me.

I didn't quite understand when suggested using `type-string` - did you mean enter `=` and then do a type conversion to string but I couldn't get it to work?

It's a json dump from dynamoDB - the snag is that the nested data is small but hangs off a number of top level cells.

**Reply by saulpw** [reply]
@piersoh You can use `~` to type a column as a string.  Or, you can expand nested columns with `(` and you can then search all columns with `g/`.

**Reply by piersoh** [reply]
Ah thanks - they're both useful approaches and work well if I know which column I'm expecting to find the value. And if I'm not sure which column it is then I figured I can use `g(` and then using `g/` which is handy! Might be worth adding it to help docs?

**Reply by saulpw** [reply]
I'm sure it's in there somewhere already.  As always, the question is, where could this live (or be linked from) where you would have found it naturally?
